packages/x-components/src/x-modules/tagging/wiring.ts

```diff
@@ -93,6 +93,16 @@ export const trackResultClickedWire = filter(
   ({ eventPayload: { tagging } }) => !!tagging?.click
 );
 
+/**
+ * Tracks the tagging of the add to cart.
+ *
+ * @public
+ */
+export const trackAddToCartWire = filter(
```

Comment:
This one and `trackResultClickedWire` use the same code to filter and map, being the `tagging` property the only difference. We should extract that logic.

Reply:
I agree, these events are going to grow, so we can add a factory for this wires in this same file. Here is my proposal:

```ts
function createTrackResultWire(property: keyof Result['tagging']): Wire<Result> {
  return filter(
    mapWire(wireDispatch('track'), ({ tagging }) => tagging[property]!),
    ({ eventPayload: { tagging } }) => !!tagging?.[property]
  );
}
```